### PR TITLE
Added DataLayer support

### PR DESCRIPTION
**Pull request by j92**
- Enable Datalayer in config with 
```
data_layer_on: 'true'
```
- Push an array or just a key and a value.
- Push a key and a value in Twig with 
```
{{ GoogleDataLayerPush('test','from whit\'in twig') }}
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# gitignore for GoogleTagManager Bolt extension
+config.yml
+config.yml.dist
```
Comment (tvlooy) [inline .gitignore]: dist files are the ones that you should provide and not exclude

**Comment by tvlooy** [comment]
can you use PSR0/1 coding style for all your code?

File: GoogleDataLayer.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Bolt\Extension\Ctors\GoogleTagManager;
+
+/**
+ * Class for our Google DataLayer object.
+ *
+ * @author Joost van Driel, joostvdriel+github@gmail.com
```
Comment (tvlooy) [inline GoogleDataLayer.php]: copyrights are in git, no @author tags

File: Extension.php [hunk 3]
```diff
@@ -5,8 +5,14 @@
 
 use Bolt\Extensions\Snippets\Location as SnippetLocation;
 
+
+include_once('GoogleDataLayer.php');
```
Comment (tvlooy) [inline Extension.php]: use autoloading instead of include

**Comment by tvlooy** [comment]
why not just always enable the datalayer? I don't really see the point of adding an option for this

File: GoogleDataLayer.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+    /** @var array */
+    protected $data;
+
+    /**
+     * Default constructor
+     */
+    public function __construct()
+    {
+
```
Comment (tvlooy) [inline GoogleDataLayer.php]: no empty functions

File: GoogleDataLayer.php [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    private function arrayToString($data){
+
+        if( empty($data) ){
+            return "";
+        }
+
+        // basis van datalayer
+        $data_layer = "";
+
+        // Loop door de data array en bouw dataLayer String
```
Comment (tvlooy) [inline GoogleDataLayer.php]: can you make all comments english?

File: Extension.php (hunk 3 above)
Comment (j92) [inline Extension.php]: Thanks for the feedback. I’m currently working on it. How can I make sure the autoloading is working? When I remove the include_once, it cannot find the class GoogleDataLayer.

**Comment by j92** [comment]
You are right, there is no point in having no DataLayer. I will remove the option.

File: Extension.php (hunk 3 above)
Comment (tvlooy) [inline Extension.php]: maybe you forgot a "use Bolt\Extension\Ctors\GoogleTagManager\GoogleDataLayer;"?

Comment (j92) [inline Extension.php]: Autoloading should work fine when the namespaces are correct. And thats the case.

**Comment by tvlooy** [comment]
I'm going to test and merge this tonight.

**Comment by tvlooy** [comment]
I made a few changes. the most important one is that if you add a value twice it will overwrite the previous value. Tagged as v1.0.2

Thanks!

**Comment by j92** [comment]
Great! Thanks for the improvement and the feedback.

**Pull request merged**